fcrepo-http-api/src/test/java/org/fcrepo/http/api/FedoraLdpTest.java

```diff
@@ -787,9 +789,32 @@ public void testCreateNewBinary() throws RepositoryException, MalformedRdfExcept
         }
     }
 
+    @Test(expected = InsufficientStorageException.class)
+    public void testCreateNewBinaryWithInsufficientResources() throws RepositoryException, MalformedRdfException,
+    InvalidChecksumException,
+    InsufficientStorageException, IOException {
+        setResource(Container.class);
+        when(mockBinaryService.findOrCreate(mockSession, "/b")).thenReturn(mockBinary);
+
+
+        try (final InputStream content = toInputStream("x")) {
+
+            final RuntimeException ex = new RuntimeException(new IOException("root exception", new IOException(
+                    "No space left on device")));
+            doThrow(ex).when(mockBinary).setContent(content, APPLICATION_OCTET_STREAM_TYPE.toString(),
+                    Collections
+                    .emptySet(),
+                    "", null);
+
+            final Response actual = testObj.createObject(null, APPLICATION_OCTET_STREAM_TYPE, "b", content, null,
+                    null);
+            verify(mockBinary).setContent(content, APPLICATION_OCTET_STREAM, Collections.emptySet(), "", null);
```

Comment:
Does this line ever get invoked? It seems like the exception will be thrown at line `809`.